aiounifi/models/voucher.py

```diff
@@ -0,0 +1,216 @@
+"""Hotspot vouchers as part of a UniFi network."""
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Self, TypedDict
+
+from .api import ApiItem, ApiRequest
+
+
+class TypedVoucher(TypedDict):
+    """Voucher type definition."""
+
+    _id: str
+    site_id: str
+    note: str
+    code: str
+    quota: int
+    duration: float
+    qos_overwrite: bool
+    qos_usage_quota: str
+    qos_rate_max_up: int
+    qos_rate_max_down: int
+    used: int
+    create_time: float
+    start_time: float
+    end_time: float
+    for_hotspot: bool
+    admin_name: str
+    status: str
+    status_expires: float
```

Comment:
You have a lot of attributes which seems to be optional in `class Voucher(ApiItem):` if they are expected to be omitted in certain cases please mark those with NotRequired

Reply:
Good advice. Done.